server/index.js

```diff
@@ -65,15 +66,25 @@ app.get('/tetraflix/genre/:genre', (req, res) => {
 
 app.post('/tetraflix/sessionData', (req, res) => {
   const { events } = req.body;
+  const movies = [];
   events.forEach((event) => {
+    movies.push([event.movie.id, event.progress]);
     if (event.progress === 1) {
       postgresDb.Movie.increment('views', { where: { id: event.movie.id } })
         .catch((err) => {
           throw err;
         });
     }
   });
-  res.sendStatus(201);
+  // calculateCW(req.body.userId, movies)
+  //   .then(() => res.sendStatus(201))
+  //   .catch((err) => {
+  //     throw err;
+  //   });
+  calculateCW(req.body.userId, movies);
```

Comment:
Instead of using setTimeout, how about wait to return a response until calculateCW is completed?  You can have calculateCW return (and inside calculateCW, query.select return) to make sure all CW list update is complete before returning response status code 201.

Reply:
Now that the calculateCW function returns a promise, I have refactored this to not use setTimeout.